crates/ra_hir_def/src/nameres/raw.rs

```diff
@@ -197,6 +206,18 @@ pub(super) struct MacroData {
     pub(super) builtin: bool,
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub(super) struct AttributeMacro(RawId);
+impl_arena_id!(AttributeMacro);
+
+#[derive(Debug, PartialEq, Eq)]
+pub(super) struct AttributeMacroData {
+    pub(super) ast_id: FileAstId<ast::ModuleItem>,
+    pub(super) path: Path,
+    /// Whether this is a derive invocation. If so, `path` is just the trait path.
+    pub(super) derive: bool,
+}
```

Comment:
I think we need to treat any unknown attribute as a macro invocation anyway. I'll try whether it's simpler to do it in the def collector, though.

Reply:
> we need to treat any unknown attribute as a macro invocation anyway

Reply:
> we need to treat any unknown attribute as a macro invocation anyway

Hmm, we might be able to figure out that some attributes are inert using name resolution though, so that's a good point :thinking:

Reply:
Yeah, that works, actually...